src/pip/_internal/vcs/versioncontrol.py

```diff
@@ -421,6 +421,18 @@ def get_url_rev_and_auth(cls, url: str) -> Tuple[str, Optional[str], AuthInfo]:
                     "which is not supported. Include a revision after @ "
                     "or remove @ from the URL.".format(url)
                 )
+        # check for typos in frag
+        parameter_names = urllib.parse.parse_qs(frag).keys()
+        if not all([p in ("egg", "subdirectory") for p in parameter_names]):
```

Comment:
```suggestion
        unknown_param_names = {
            key
            for key in urllib.parse.parse_qs(frag)
            if key not in ("egg", "subdirectory")
        }
        if unknown_param_names:
```

as a bonus we can show those unknown params in the error message.